susemanager-utils/susemanager-sls/src/beacons/pkgset.py

```diff
@@ -49,20 +55,25 @@ def beacon(config):
           pkgset:
             interval: 5
 
-    '''
-
-    global COOKIE_PATH
+    """
 
     ret = []
-    if os.path.exists(COOKIE_PATH):
-        with open(COOKIE_PATH) as ck_file:
+    for cookie_path in PKGSET_COOKIES:
+        if not os.path.exists(cookie_path):
+            continue
+        with open(cookie_path) as ck_file:
             ck_data = ck_file.read().strip()
             if __virtualname__ not in __context__:
-                __context__[__virtualname__] = ck_data
+                cache_data = CACHE.fetch("beacon/pkgset", "cookie").get("data", None)
+                if cache_data:
+                    __context__[__virtualname__] = cache_data
+                else:
+                    __context__[__virtualname__] = ck_data
+                    CACHE.store("beacon/pkgset", "cookie", {"data": ck_data})
```

Comment:
I'm adding some comments as suggestions here that maybe help clarifying a bit the logic of the code.

```suggestion
                # After a minion restart, when this is running for first time, there is nothing in context yet
                # So, if there is any data in the cache, we put it in the context, if not we put the new data.
                # and update the data in the cache.
                cache_data = CACHE.fetch("beacon/pkgset", "cookie").get("data", None)
                if cache_data:
                    __context__[__virtualname__] = cache_data
                else:
                    __context__[__virtualname__] = ck_data
                    CACHE.store("beacon/pkgset", "cookie", {"data": ck_data})
```